### PR TITLE
LPD-18884 add functional tests for scriptElementsAttributes in globalJS client extension

### PR DESCRIPTION
**Pull request by thiago-buarqque**
Adding functional tests as described in [LPD-18884](https://liferay.atlassian.net/browse/LPD-18884)

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:playwright

**Review by ethib137 — CHANGES_REQUESTED** [review]
Looking good... just a few changes.

File: modules/test/playwright/tests/client-extension-web/js.spec.ts [hunk 1]
```diff
@@ -83,3 +86,242 @@ test('JS client extension does not allow "src" as a script element attribute', a
...
 });
+
+const assertDefaultSelectedLoadType = async (
+	clientExtensionName: string,
+	page: Page,
+	loadType: string
+) => {
+	const loadTypeSelector = page
+		.locator(`text=${clientExtensionName}`)
+		.locator('xpath=../td/select');
```
[the remark this answers is not in the document]
Comment (thiago-buarqque) [reply modules/test/playwright/tests/client-extension-web/js.spec.ts:97]: Done, changed to
```
const loadTypeSelector = page
		.locator(`tr`, {hasText: clientExtensionName})
		.locator(".load-type-select");
```

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:playwright

**Comment by thiago-buarqque** [comment]
Hey @ethib137, the playwright errors are unrelated. If my changes are fine, please forward to Brian

File: modules/test/playwright/tests/client-extension-web/js.spec.ts [hunk 2]
```diff
@@ -83,3 +86,242 @@ test('JS client extension does not allow "src" as a script element attribute', a
...
 	expect(page.getByText('Use the "JavaScript URL" field.')).toBeVisible();
 });
+
+const assertDefaultSelectedLoadType = async (
+	clientExtensionName: string,
+	page: Page,
+	loadType: string
+) => {
+	const loadTypeSelector = page
+		.locator(`tr`, {hasText: clientExtensionName})
```
Comment (ethib137) [inline modules/test/playwright/tests/client-extension-web/js.spec.ts:96]: The `tr` should be wrapped with normal single quotes.

File: modules/test/playwright/tests/client-extension-web/js.spec.ts [hunk 3]
```diff
@@ -83,3 +86,242 @@ test('JS client extension does not allow "src" as a script element attribute', a
...
+};
+
+type TScriptAttribute = {
+	name: string;
+	type: 'boolean' | 'string';
+	value: string;
+	valueWhenInPage: string | null;
+};
+
+type TGlobalJSWithAttributesTest = {
```
Comment (ethib137) [inline modules/test/playwright/tests/client-extension-web/js.spec.ts:111]: This should match the method name: `TGlobalJSClientExtensionWithAttributes`

**Review by ethib137 — CHANGES_REQUESTED** [review]
Just a couple minor changes. Thanks Thiago!

**Comment by thiago-buarqque** [comment]
ci:test:playwright

**Comment by ethib137** [comment]
ci:forward

**Review by ethib137 — APPROVED** [review]
LGTM

**Pull request closed without merging**